File: CS50x/2. Arrays/5.c
/*

> Date Created: July 20, 2023
> Author: Ishaan Rastogi
> Purpose: User input version of 5.c
> Operating System: This is only for Windows OS, it may or may not work on other OS
> Program Status: 100% Working

/*

Array- A way of storing your data back to back in computer's memory in such a way that you can access each individual member easily.

*/

#include<stdio.h>
#include"cs50.h"

int main()
{

    int scores[3]; //array
    //loop for user input of scores
    for (int i=0; i < 3; i++ )
    {
    scores[i] = get_int("Score:");
    }

    printf("Average: %f\n", (scores[0]+ scores[1] + scores[2] ) / (float) 3); // Or you can explicitly convert 3 into float using (float)

}

/* 

Ctrl + Shift + `

Terminal Code Loading...

gcc 5.c -o 5 cs50.c
./5

NOTE- Whenever you are using cs50 functions like get_string, make sure to follow the YouTube video named 'how to use cs50 library in vscode' till
the end, uploaded by a channel named Mahmoud Saad in the year 2022. Don't miss any single second.

*/

File: Web Development Projects/15 Intermediate JS/5.js
/*
> Date Created: June 10, 2024
> Author: Ishaan Rastogi
> Purpose: Leap Year Calculator
> Operating System: This is only for Windows OS, it may or may not work on other OS
> Program Status: 100% Working
*/

/*

Leap year is a year:

1. that is evenly divisible by 4.
2. except year that is evenly divisible by 100              Eg- year 2100 is not an leap year
3. unless the year is also evenly divisible by 400.         Eg- year 2400 is a leap year
*/

function isLeap(year) {
    if (year % 4 === 0) {
        if (year % 100 === 0) {
            if (year % 400 === 0) {

                return "Leap year"

            } else { return "Not leap year." }
        } else { return " Leap year. " }
    } else { return "Not leap year." }
}

/* 
For Terminal Code... Ctrl + Shift + `
*/